The numbers of cpu should be larger than 0.

There is a floating point error because the number of cpus
was sometimes incorrectly returned as 0 instead of 1. So
if the the number of cpu is read as 0, it should be
replaced by 1.
 
Signed-off-by: Jiang, Bin <bin.jiang@windriver.com>

3/14/2014
Take it from WRLinux 4.3 for monit 4.9

Upstream-Status: Pending

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 process/sysdep_LINUX.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/process/sysdep_LINUX.c
+++ b/process/sysdep_LINUX.c
@@ -142,6 +142,9 @@ int init_process_info_sysdep(void) {
     return FALSE;
   }
 
+  if (num_cpus == 0)
+    num_cpus = 1;
+
   if((page_size = sysconf(_SC_PAGESIZE)) <= 0) {
     DEBUG("system statistic error -- cannot get page size: %s\n", STRERROR);
     return FALSE;
